HKMP/Game/Client/Entity/MantisLord.cs

```diff
@@ -0,0 +1,907 @@
+                    }
+                    _throneFsm.ExecuteActions("Stand", 1);
+                    _throneFsm.ExecuteActions("Cage", 1, 2, 3, 4, 5);
+                    _throneFsm.ExecuteActions("Floors", 1);
+
+                    _throneFsm.SetState("Floors");
+                }
+
+                if (state == ThroneState.Empty) {
+                    if (!_godHome) {
+                        _challengePromptFsm.ExecuteActions("Send Challenge", 1);
+                        _throneFsm.ExecuteActions("Music", 2, 3);
+                    }
+                    _throneFsm.ExecuteActions("Cage", 1, 2, 3, 4, 5);
+                    _throneFsm.ExecuteActions("Floors", 1);
+                    _throneFsm.ExecuteActions("Leave", 1, 2);
+                    _throneFsm.ExecuteActions("Wake", 1, 2);
+
+                    _throneFsm.SetState("Wake");
+                }
+
+                if (state == ThroneState.Defeated) {
+                    if (!_godHome) {
+                        _challengePromptFsm.ExecuteActions("Send Challenge", 1);
+                        _throneFsm.ExecuteActions("Music", 2, 3);
+                    }
+                    _throneFsm.ExecuteActions("Cage", 1, 2, 3, 4, 5);
+                    _throneFsm.ExecuteActions("Floors", 1);
+                    _throneFsm.ExecuteActions("Defeated", 1);
+
+                    _throneFsm.SetState("Defeated");
+                }
+            }
+
+            // Now call the base class initializer, as we might have activated the mantis lord object
+            base.InternalInitializeAsSceneClient(baseStateIndex);
+        }
+
+        protected override void InternalSwitchToSceneHost() {
+            base.InternalSwitchToSceneHost();
+
+            RestoreAllTransitions(_throneFsm);
+            if (!_godHome) {
+                RestoreAllTransitions(_challengePromptFsm);
+            }
+
+            switch (_lastThroneAnimation) {
+                case ThroneAnimation.Music:
+                    _throneFsm.SetState("Stand");
+                    break;
+                case ThroneAnimation.Stand:
+                    _throneFsm.SetState("Cage");
+                    break;
+                case ThroneAnimation.Cage:
+                    _throneFsm.SetState("Floors");
+                    break;
+                case ThroneAnimation.Floors:
+                    _throneFsm.SetState("Leave");
+                    break;
+                case ThroneAnimation.Leave:
+                    _throneFsm.SetState("Wake");
+                    break;
+                case ThroneAnimation.Wake:
+                    var healthManager = GameObject.GetComponent<HealthManager>();
+                    // We might have never received the defeated event, but the entity is dead already
+                    if (healthManager.GetIsDead()) {
+                        _throneFsm.SetState("Pause");
+                    }
+                    else {
```

Comment:
Unnecessary line break.
```suggestion
                    } else {
```